packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -372,16 +372,20 @@ describe('<DataGrid /> - Layout & Warnings', () => {
       });
     });
 
+    it('Grid wrapper should have height different from 0 when autoHeight', () => {
+      const { getByRole, getAllByRole } = render(
+        <div style={{ width: 300 }}>
+          <DataGrid {...baselineProps} autoHeight />
+        </div>,
+      );
 
-      it('Grid wrapper should have height different from 0 when autoHeight', () => {
-        const { getByRole } = render(
-          <div style={{ width: 300 }}>
-            <DataGrid {...baselineProps} autoHeight />
-          </div>,
-        );
+      const colHeaderRowHeight = getAllByRole('row')[0].clientHeight;
+      const totalRowsHeight = getAllByRole('row')[1].clientHeight * 3;
+      const footerHeight = document.querySelector('.MuiDataGrid-footer')?.clientHeight;
 
-        expect(getByRole('grid').clientHeight).to.not.equal(0);
-      });
+      expect(getByRole('grid').clientHeight).to.equal(
+        colHeaderRowHeight + totalRowsHeight + footerHeight + 1, // The 1 is to compensate, needs to be fixed
```

Comment:
1 => Border?

Reply:
I've replaced it with what @oliviertassinari suggested. Idk why there was a mismatch of 1px (if it was border then it would have been at least 2px, top, and bottom).